fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -70,30 +89,53 @@ def __init__(self, fapd_control_enabled=True):
             self.fapd_profiling_stderr = None
 
     def start(self, instance=FapdMode.ONLINE):
-        # ToDo: Add logic to that online and profiling instances are mutex
-        self.mode = instance
-        self._start()
+        # If profiling, get current fapd state, and stop it if running
+        if instance == FapdMode.PROFILING:
+            self.capture_online_state()
+            self._stop()
+        else:
+            # Check status of fapd profiling instance and bring it down
+            if self._fapd_profiling_status:
+                logging.debug("FapdManager::start(): Shutting down prof fapd")
+                self._stop(FapdMode.PROFILING)
```

Comment:
This can be simplified to an `elif` statement.

Reply:
Done